Packs/Cyberpion/Integrations/Cyberpion/Cyberpion.py

```diff
@@ -0,0 +1,530 @@
+import demistomock as demisto  # noqa: F401
+from typing import Dict, Tuple, List
+from datetime import timezone
+
+from CommonServerPython import *
+
+"""Cyberpion Integration for Cortex XSOAR (aka Demisto)
+"""
+
+''' IMPORTS '''
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%d %H:%M:%S.%f %Z'
+
+MAX_PAGES_TO_GRAB = 5
+DEFAULT_MAX_INCIDENTS_TO_FETCH = 200
+CONNECTION_TIMEOUT = 30.0
+READ_TIMEOUT = 30.0
+VALID_STATUS_CODES = (200,)
+NUM_OF_RETRIES = 3
+BACKOFF_FACTOR = 1.0  # see documentation in CommonServerPython._http_request
+ACTION_ITEM_TYPE_NAME = 'cyberpion_action_item'
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any Demisto logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def get_connections(self, domain: str, reverse: bool = False):
+        connection_direction = 'to_domain' if reverse else 'from_domain'
+        params = {connection_direction: domain}
+        http_responses = []
+        page_num = 1
+        while page_num <= MAX_PAGES_TO_GRAB:
+            # call API
+            params['page'] = str(page_num)
+            demisto.info(f'getting connections, page num: {page_num}, domain={domain}')
+            http_response = self._http_request(
+                method='GET',
+                url_suffix='/connections/',
+                params=params,
+                resp_type='json',
+                ok_codes=VALID_STATUS_CODES,
+                timeout=(CONNECTION_TIMEOUT, READ_TIMEOUT),
+                retries=NUM_OF_RETRIES,
+                backoff_factor=BACKOFF_FACTOR,
+                raise_on_status=True
+            )
+            demisto.info(f'after getting connections, page num: {page_num}, domain={domain}')
+            if 'results' not in http_response:
+                raise Exception('failed to read connections.\nError: got response without \'results\' key')
+            http_responses.append(http_response['results'])
+            # for idx, action_items_result in enumerate(http_response['results']):
+            #     technical_det = action_items_result.get('technical_details', {})
+            #     http_response['results'][idx]['technical_details'] = '\n'.join(
+            #         ['{}: {}'.format(k.strip(), v.strip()) for k, v in technical_det.items()] if type(
+            #             technical_det) is dict else technical_det.strip())
+            #     http_response['results'][idx]['alert_type'] = ACTION_ITEM_TYPE_NAME
+            if not http_response.get('next'):
+                break
+            else:
+                time.sleep(2)
```

Comment:
We do not allow using `sleep` in the integration code as it might lead to performance issues - why is it needed if we already use the backoff factor?

Reply:
We don't want the server to throttle the request. The sleep happens when getting multiple pages at once, not when retrying (then we use backoff factor)

Reply:
we won't be able to use `sleep` in the code
we should either use the backoff factor or come up with another solution on your end